trapper/pipelines/question_answering_pipeline.py

```diff
@@ -188,112 +188,105 @@ def __init__(
+        return {"start": start, "end": end, **model_inputs}
 
-                for start_token_ind, end_tok_ind, score in zip(
-                    starts, ends, scores
-                ):
-                    answers.append(
-                        {
-                            "score": score.item(),
-                            "answer": self._construct_answer(
-                                example["context"],
-                                input_ids,
-                                start_token_ind,
-                                end_tok_ind,
-                            ),
-                        }
-                    )
+    def postprocess(
+        self,
+        model_outputs,
+        handle_impossible_answer=False,
+        topk=1,
+        max_answer_len=15,
+    ):
+        start = model_outputs["start"]
+        end = model_outputs["end"]
+        example = model_outputs["example"]
+        indexed_instance = model_outputs["indexed_instance"]
+        single_instance_batch = self._data_collator.build_model_inputs(
+            (indexed_instance,), should_eliminate_model_incompatible_keys=False
+        )
+        self._data_collator.pad(single_instance_batch)
+        input_ids = np.asarray(single_instance_batch["input_ids"][0])
+
+        min_null_score = self._MIN_NULL_SCORE
+        answers = []
+        for (start_, end_) in zip(start, end):
+            # Normalize logits and spans to retrieve the answer
+            start_ = self.normalize_logits(start_).reshape(1, -1)
+            end_ = self.normalize_logits(end_).reshape(1, -1)
 
-            if kwargs["handle_impossible_answer"]:
+            if handle_impossible_answer:
+                min_null_score = min(min_null_score, (start_[0] * end_[0]).item())
+
+            # Mask BOS and EOS tokens
+            start_[0, 0] = end_[0, 0] = 0.0
+
+            starts, ends, scores = self._decode(start_, end_, topk, max_answer_len)
+
+            for start_token_ind, end_tok_ind, score in zip(starts, ends, scores):
                 answers.append(
                     {
-                        "score": min_null_score,
-                        "answer": convert_spandict_to_spantuple(
-                            {"start": 0, "text": ""}
+                        "score": score.item(),
+                        "answer": self._construct_answer(
+                            example["context"],
+                            input_ids,
+                            start_token_ind,
+                            end_tok_ind,
                         ),
                     }
                 )
 
-            answers = self._get_topk_answers(answers, kwargs)
-            all_answers.append(answers)
+        if handle_impossible_answer:
+            answers.append(
+                {
+                    "score": min_null_score,
+                    "answer": convert_spandict_to_spantuple(
+                        {"start": 0, "text": ""}
```

Comment:
I'm not 100% sure on this so I'll tag @cemilcengiz here, but I guess it'd be good to assure consistency, so I think we need to change this as

```python
{"start": -1, "text": ""}
```